pkg/ccl/backupccl/create_scheduled_backup.go

```diff
@@ -128,9 +125,74 @@ func setScheduleOptions(
 	return nil
 }
 
+type scheduleRecurrence struct {
+	cron      string
+	frequency time.Duration
+}
+
+var neverRecurs *scheduleRecurrence
+
+func computeScheduleRecurrence(
+	now time.Time, evalFn func() (string, error),
+) (*scheduleRecurrence, error) {
+	if evalFn == nil {
+		return neverRecurs, nil
+	}
+	cron, err := evalFn()
+	if err != nil {
+		return nil, err
+	}
+	expr, err := cronexpr.Parse(cron)
+	if err != nil {
+		return nil, err
+	}
+	nextRun := expr.Next(now)
+	frequency := expr.Next(nextRun).Sub(nextRun)
+	return &scheduleRecurrence{cron, frequency}, nil
+}
+
+var humanDurations = map[time.Duration]string{
+	time.Hour:          "hour",
+	24 * time.Hour:     "day",
+	7 * 24 * time.Hour: "week",
+}
+
+func (r *scheduleRecurrence) Humanize() string {
+	if d, ok := humanDurations[r.frequency]; ok {
+		return "every " + d
+	}
+	return "every " + r.frequency.String()
+}
+
+var forceFullBackup *scheduleRecurrence
+
+func pickFullRecurrenceFromIncremental(inc *scheduleRecurrence) *scheduleRecurrence {
+	if inc.frequency <= time.Hour {
+		// If incremental is faster than once an hour, take fulls every day,
+		// some time between midnight and 1 am.
+		return &scheduleRecurrence{
+			cron:      fmt.Sprintf("%d 0 * * *", rand.Intn(60)),
```

Comment:
I'm not sure about about the rand jitter here. I know the user didn't express a preference so we're picking one for them and thus we figure we'll avoid thundering herds at midnight, but what makes us think midnight to 1AM _UTC_ is the right window? why not pick a random hour too? 

I feel like if you don't say when, we can assume you don't _care_ when, potentially including that you don't have load that makes you care / what to offset spikes. 

Also, I imagine myself creating a schedule: if I omitted the full, i'd open the resulting schedules and see it at 00:00:00 and go "oh, i didn't tell it when". If I open it and see "00:47:00" I'm going to be a bit confused.... where did 47 come from?? Did you read some unintialized input buffer? what is going on here?"

I donno, maybe @mwang1026 has thoughts